js/ui/form/ui.form.js

```diff
@@ -640,15 +640,16 @@ var Form = Widget.inherit({
     _getColCount: function($element) {
         var index = 0,
             isColsExist = true,
+            colCountByOption = this.option("colCount"),
             $cols;
 
         while(isColsExist) {
             $cols = $element.find("." + FORM_FIELD_ITEM_COL_CLASS + index);
             if(!$cols.length) {
-                isColsExist = false;
-            } else {
-                index++;
+                isColsExist = index < colCountByOption;
```

Comment:
"auto" is valid value for the `colCount` option.
Keep in mind, that `colCountByScreen` can define a various columns count for different screen sizes.
Try to pass a `colCount` that returns `layoutManager._getColCount` method to `_applyLabelsWidth` at `_alignLabelsInColumn`